crypten/nn/onnx_converter.py

```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import io
+from collections import OrderedDict
+
+import onnx
+import torch
+import torch.onnx.symbolic_helper as sym_help
+import torch.onnx.symbolic_registry as sym_registry
+import torch.onnx.utils
+from onnx import numpy_helper
+
+from . import module
+
+
+try:
+    import tensorflow as tf  # noqa
+    import tf2onnx
+
+    TF_AND_TF2ONNX = True
+except ImportError:
+    TF_AND_TF2ONNX = False
+
+
+# mapping from ONNX to crypten.nn for modules with different names:
+ONNX_TO_CRYPTEN = {
+    "AveragePool": module.AvgPool2d,
+    "BatchNormalization": module._BatchNorm,
+    "Gemm": module.Linear,
+    "MaxPool": module.MaxPool2d,
+    "Pad": module._ConstantPad,
+    "Relu": module.ReLU,
+    "ReduceMean": module.Mean,
+    "ReduceSum": module.Sum,
+}
+
+
+def from_pytorch(pytorch_model, dummy_input):
+    """
+    Static function that converts a PyTorch model into a CrypTen model.
+    """
+    # TODO: Currently export twice because the torch-to-ONNX symbolic registry
+    # only gets created on the first call.
+    with io.BytesIO() as f:
+        _export_pytorch_model(f, pytorch_model, dummy_input)
+
+    # update ONNX symbolic registry with CrypTen-specific functions
+    _update_onnx_symbolic_registry()
+
+    # export again so the graph is created with CrypTen-specific registry
+    with io.BytesIO() as f:
+        f = _export_pytorch_model(f, pytorch_model, dummy_input)
+        f.seek(0)
```

Comment:
It would be great if the logic here is put into a separate function. An interesting use-case for PySyft is where we can serialize a pytorch model into this bytes object, send it through the network, then build the crypten model in another machine.

Reply:
Great, happy to restructure. I'll update the pull request.